doc/feature/transferFunctions.md

```diff
@@ -0,0 +1,238 @@
+Transfer functions {#transferFunctions}
+============
+
+Being able to specify a mapping between scalar variables and parameters used
+for represent visually those variable is a recurring theme in visualization
+applications. An exemplary case is the transfer functions that map density to
+RGA and opacity values in volume rendering with an emission-absorption optical
+model.
+
+This specification describes the messages and C++ helper classes to deal with
+transfer functions (TFs), considering the usual RGBA as a particular case.
+
+## Requirements
+
+- RGBA colormaps need to be considered first class citizens.
+- However, some applications require more attributes that just RGBA, so it
+  should be possible to describe TFs that map a variable to arbitrary attributes.
+- The C++ helper class will encapsulate a colormap and allow the conversion of
+  the network representation into a OpenGL ready lookup table for creating a
+  texture.
+- The network representation shouldn't restrict the options in the
+  implementation of a GUI editor.
+
+## Design
+
+The consensus has been opting for using 2D piece-wise linear functions specified
+as (unsorted) lists of control points in R^2. An RGBA colormap is an aggregation
+of 4 of this control point lists. Generic TFs are described with a list of
+control points and the name of the target parameter.
+
+Control points are always in the absolute scale, that means that the x coordinate
+refers to the scale of the source variable and the y coordinate refers to the
+scale of the target attribute. The range of RGBA channels will be [0, 1] unless
+an application specifies something different (e.g. in physically based TFs)
+For example, a simple black to white colormap
+for neuron membrane voltage
+could be described by the control points (-80, 0), (-10, 1)
```

Comment:
Imo the ColorMap should always be normalized, otherwise its semantics are left open. Your example illustrates my point: you are describing a neuron simulation voltage colormap. My gut feeling is the range is separate.